packages/wdio-config/src/lib/ConfigParser.ts

```diff
@@ -348,6 +348,7 @@ export default class ConfigParser {
                 log.error('Unexpected depth of hierarchical arrays')
             } else {
                 pattern = pattern.toString().replace(/\\/g, '/')
+                pattern = pattern.toString().replace(/[[\]]/g, '\\$&')
```

Comment:
I think you missed the escape for the opening bracket
```suggestion
                pattern = pattern.toString().replace(/[\[\]]/g, '\\$&')
```

Reply:
> Can we add a test for this?
> 
> > Getting 'is-stream' issue while running unit tests.
> 
> Let's log this as a different issue.

What do you think can be the fix? Downgrading tempy?

Reply:
let's discuss this in #8671